src/main/scala/ignition/core/jobs/utils/SparkContextUtils.scala

```diff
@@ -1,21 +1,26 @@
 package ignition.core.jobs.utils
 
-import java.util.Date
-
 import ignition.core.utils.ByteUtils
+import ignition.core.utils.CollectionUtils._
+import org.apache.hadoop.conf.Configuration
 import org.apache.hadoop.io.LongWritable
-import org.apache.spark.SparkContext
+import org.apache.hadoop.io.compress.CompressionCodecFactory
+import org.apache.spark.{Partitioner, SparkContext}
 import org.apache.hadoop.fs.{FileStatus, Path, FileSystem}
 import org.apache.spark.rdd.{UnionRDD, RDD}
-import org.joda.time.{DateTimeZone, DateTime}
+import org.joda.time.DateTime
 import ignition.core.utils.DateUtils._
 
+import scala.collection.mutable.ArrayBuffer
+import scala.io.Source
 import scala.reflect.ClassTag
 import scala.util.Try
 
-
 object SparkContextUtils {
 
+  case class Bucket(var size: Long, paths: ArrayBuffer[String])
```

Comment:
Nothing say that a case class should be immutable. But is something that I really didn't expect. I should try to remove its multability or at least remove its case.